src/math/Box3.js

```diff
@@ -209,21 +209,38 @@ class Box3 {
 
 		object.updateWorldMatrix( false, false );
 
-		const geometry = object.geometry;
+		if ( object.boundingBox !== undefined ) {
 
-		if ( geometry !== undefined ) {
+			if ( object.boundingBox === null ) {
 
-			if ( geometry.boundingBox === null ) {
-
-				geometry.computeBoundingBox();
+				object.computeBoundingBox();
 
 			}
 
-			_box.copy( geometry.boundingBox );
+			_box.copy( object.boundingBox );
 			_box.applyMatrix4( object.matrixWorld );
 
 			this.union( _box );
 
+		} else {
+
+			const geometry = object.geometry;
+
+			if ( geometry !== undefined ) {
+
+				if ( geometry.boundingBox === null ) {
+
+					geometry.computeBoundingBox();
+
+				}
+
+				_box.copy( geometry.boundingBox );
```

Comment:
I'm afraid this make the code not more readable since the union operation should only happen for a 3D objects that have bounding volumes on object level _or_ a geometry with valid bounding volumes. You would need a separate `if` statement after the first one to implement this. To me, it's seems simpler to inline the copy/transform/union in the existing if statements.